win_lgpo registry.pol encoding updates

lomeroe opened the pull request:
### What does this PR do?
Updates win_lgpo to not decode the registry.pol file wholesale, but instead decode the items of the file as necessary while processing.  Some columns of a registry entry where not encoded properly (i.e. the size and type fields were a utf-16-le encoded number and a null character as opposed to being utf-32-le encoded to have the proper raw length in the registry.pol file).

Uses byte objects for all registry.pol items/etc to be py2/py3 compatible.

See https://msdn.microsoft.com/en-us/library/cc232696.aspx for additional information on the encodings of the "columns" of a registry.pol entry.

Also fixes an issue in get_policy_info where items from the static _policy_info class would not be found if the short name was used (i.e. the key name in the policies dict)

### What issues does this PR fix or reference?
#44516 - this should be a more complete fix

@twangboy - please test this out on your end, my tests came back successful on both 2016.11 and 2017.7 py2 and py3 (though this will need to be properly backported to 2016.11 and to develop).

### Previous Behavior
Encoding errors would be returned on py3 for items whose data item where a unicode surrogate when attempting to encode, the same would happen on py2 when attempting to decode a surrogate value that existed in the registry.pol file.

### New Behavior
Data items are not encoded/decoded unless necessary resulting in no need to ignore surrogates/etc (i.e. data is compared as byte objects instead of converting to more human friendly/readable objects)

### Tests written?
No

### Commits signed with GPG?
No

Please review [Salt's Contributing Guide](https://docs.saltstack.com/en/latest/topics/development/contributing.html) for best practices.

See GitHub's [page on GPG signing](https://help.github.com/articles/signing-commits-using-gpg/) for more information about signing commits with GPG.

lomeroe committed:
do not decode registry.pol file wholesale, but instead decode individual
elements of the file

also fix issue where the get_policy_info function would not find
policies in the policy_data class by their "short" name (i.e. the key
name in the dict)

twangboy reviewed (approved):
This is Awesome! This should fix a few other key issues we're having where `_read_regpol_path` was dying because it was trying to decode values that weren't valid unicode.

rallytime commented:
@lomeroe There are some lint failures on this: https://jenkins.saltstack.com/job/PR/job/salt-pr-lint-n/17265/violations/file/salt/modules/win_lgpo.py/

Can you fix those?

lomeroe committed:
lint fixes

encode some of the more used characters in variables and re-use instead
of re-encoding them over and over

remove a line that was commented out

rallytime commented:
@lomeroe There's still some lint errors on this one. Can you take another look?

https://jenkins.saltstack.com/job/PR/job/salt-pr-lint-n/17277/violations/file/salt/modules/win_lgpo.py/

lomeroe committed:
lint fixes for static regexes

twangboy commented:
@rallytime Looks good
